bears/generate_package.py

```diff
@@ -0,0 +1,123 @@
+import argparse
+import glob
+import os
+import shutil
+from string import Template
+import subprocess
+import sys
+
+from bears import VERSION
+from coalib.collecting.Importers import iimport_objects
+
+
+def touch(file_name):
+    """
+    Creates an empty file. An existing file remains untouched.
+
+    :param file_name: Name of the file.
+    """
+    open(file_name, 'a').close()
+
+
+def create_file_from_template(template_file, output_file, substitution_dict):
+    """
+    Creates a file from a template file, using a substitution dict.
+
+    :param template_file:     The template file.
+    :param output_file:       The file to be written.
+    :param substitution_dict: The dict from which the substitutions are taken.
+    """
+    with open(template_file) as fl:
+        template = fl.read()
+    template = Template(template).safe_substitute(**substitution_dict)
+
+    with open(output_file, 'w') as output_handle:
+        output_handle.write(template)
+
+
+def create_file_structure_for_packages(root_folder, file_to_copy, object_name):
+    """
+    Creates a file structure for the packages to be uploaded. The structure
+    will be ``root_folder/object_name/object_name/file_to_copy``.
+    The file structure has two object_name folders because ``setup.py`` file
+    needs files to be one level deeper than itself so that they can be uploaded.
+
+    :param root_folder:  The folder in which the packages are going to be
+                         generated.
+    :param file_to_copy: The file that is going to be generated the package for.
+    :param object_name:  The name of the object that is inside the file_to_copy.
+    """
+    upload_package_folder = os.path.join(root_folder, object_name, object_name)
+    os.makedirs(upload_package_folder, exist_ok=True)
+    touch(os.path.join(upload_package_folder, '__init__.py'))
+    shutil.copyfile(file_to_copy, os.path.join(upload_package_folder,
+                                               object_name + '.py'))
+
+
+def perform_upload(path):
+    """
+    Uploads the directory to PyPi, creating a ``sdist`` and a ``bdist_wheel``,
+    after registering it.
```

Comment:
you don't always want to register it, only at the first time! Registering should be a flag like `coala-bears --register --upload` would register *and* upload it